pandas/core/indexes/datetimes.py

```diff
@@ -1776,7 +1776,8 @@ def insert(self, loc, item):
 
         if isinstance(item, (datetime, np.datetime64)):
             self._assert_can_do_op(item)
-            if not self._has_same_tz(item):
+            if not self._has_same_tz(item) and item is not self._na_value:
```

Comment:
``notnull(item)``

Reply:
Will change, but FYI they are equivalent here.  notnull(item) was checked above in which case it was replaced with self._na_value.